File: trunk/poll/sharpteam1/Ilsrep.PollApplication.Protocol/Model/PollPacket.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
using System.Xml.Serialization;

namespace Ilsrep.PollApplication.Model
{
    /// <summary>
    /// Root elements that is transfered between poll client, editor and server
    /// </summary>
    [XmlRoot("pollpacket"), Serializable]
    public class PollPacket
    {
        /// <summary>
        /// Request element is used to query server
        /// </summary>
        [XmlElement("request")] public Request request = null;
        /// <summary>
        /// PollSessionList is sent on GET_LIST request
        /// </summary>
        [XmlElement("pollsessionlist")] public PollSessionList pollSessionList = null;
        /// <summary>
        /// PollSession is sent on GET_POLLSESSION request
        /// </summary>
        [XmlElement("pollsession")] public PollSession pollSession = null;
        /// <summary>
        /// Results list is sent when user finishes poll in poll client
        /// </summary>
        [XmlElement("resultslist")] public ResultsList resultsList = null;
    }
}

File: poll/jteam/src/ilsrep/poll/common/protocol/Pollpacket.java
package ilsrep.poll.common.protocol;


import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

import ilsrep.poll.common.model.Pollsession;
import ilsrep.poll.common.protocol.Answers;
import ilsrep.poll.common.protocol.User;
/**
 * The "pollpacket" element.
 * 
 * @author TKOST
 * 
 */
@XmlRootElement(name = "pollpacket")
public class Pollpacket {

    /**
     * Client's request to server.
     */
    protected Request request = null;

    /**
     * Server's response for client's request of pollsession list stored on
     * server.
     */
  

    protected Pollsessionlist pollsessionList = null;
     /**
     * A set of answer data
     */
    
      protected Answers answerlist = null;
      
      protected User user=null;

    /**
     * Used when:
     * <ul>
     * <li>server sends poll xml for client's request</li>
     * <li>editor sends xml created by him to server</li>
     * </ul>
     */
    protected Pollsession pollsession = null;

    /**
     * @see #request
     */
    public Request getRequest() {
        return request;
    }

    /**
     * @see #request
     */
    public void setRequest(Request request) {
        this.request = request;
    }

    /**
     * @see #pollsessionList
     */
    @XmlElement(name = "pollsessionlist")
    public Pollsessionlist getPollsessionList() {
        return pollsessionList;
    }

    /**
     * @see #pollsessionList
     */
    public void setPollsessionList(Pollsessionlist pollsessionList) {
        this.pollsessionList = pollsessionList;
    }

    /**
     * @see #pollsession
     */
    public Pollsession getPollsession() {
        return pollsession;
    }

    /**
     * @see #pollsession
     */

    public void setPollsession(Pollsession pollsession) {
        this.pollsession = pollsession;
    }
/**
     * @see #answerlist
     */
    @XmlElement(name = "resultslist")
    public void setResultsList(Answers answer) {
        this.answerlist = answer;
    }
/**
     * @see #answerlist
     */
    public Answers getResultsList() {
        return answerlist;
    }

/**
     * @see #user
     */
    @XmlElement(name = "user")
    public void setUser(User user) {
        this.user = user;
    }

    public User getUser() {
        return user;
    }
}
